python/across_api/auth/api.py

```diff
@@ -75,18 +79,47 @@ async def __call__(self, request: Request) -> None:
                 raise HTTPException(
                     status_code=401, detail=f"Authentication error: {e}"
                 )
+
+            return access_token
+
         else:
             raise AssertionError("No credentials passed.")
 
 
-security = JWTBearer(
+JWTBearerSecurity = JWTBearer(
     scheme_name="ACROSS API Authorization",
     description="Enter your access token.",
 )
-JWTBearerDep = [Depends(security)]
 
 
-@app.get("/auth/verify", dependencies=JWTBearerDep)
+async def ScopeAuthenticate(
+    security_scopes: SecurityScopes,
+    access_token: Annotated[dict, Depends(JWTBearerSecurity)],
+):
+    valid = False
+    scopes = access_token.get("scope", "")
+
+    # assuming the jwt scopes will be comma separated
+    token_scopes = scopes.split(",")
+
+    # check if user scopes is in endpoint scope
+    if len(security_scopes.scopes):
+        for role in token_scopes:
+            if role in security_scopes.scopes:
+                valid = True
+    else:  # endpoint has no scopes
+        valid = True
```

Comment:
```suggestion
    valid = all(scope in security_scopes.scopes for scope in token_scopes)
```

I take it that `token_scopes` are the scopes that the decorator states that the token must have, and `security_scopes.scopes` are the scopes that are actually present in the token?

Reply:
I agree with your example, but I think that the correct implementation is https://github.com/nasa-gcn/gcn.nasa.gov/pull/1874#discussion_r1479072676.